src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -76,8 +95,25 @@ QuickMarcEditor.propTypes = {
   submitting: PropTypes.bool,
   pristine: PropTypes.bool,
   initialValues: PropTypes.object,
+  form: PropTypes.shape({
+    mutators: PropTypes.object.isRequired,
+  }),
 };
 
 export default stripesFinalForm({
   navigationCheck: true,
+  mutators: {
+    addRecord: ([{ fields, index }], state, tools) => {
```

Comment:
you should use records values from `state` as `fields` you passed has initial values

in your implementation you override user input